src/main/java/emissary/place/CoordinationPlace.java

```diff
@@ -220,49 +215,124 @@ protected List<IBaseDataObject> coordinate(IBaseDataObject d, boolean hd) {
+        processForm(d, errorOccurred);
 
-            // Clean up my proxies
-            nukeMyProxies(d);
+        sproutHook(d, hd, sproutCollection);
 
-        }
+        // Allow derived classes a shot to clean up the parent
+        cleanUpHook(d);
 
+        return sproutCollection;
+    }
+
+    private void sproutHook(IBaseDataObject d, boolean hd, List<IBaseDataObject> sproutCollection) {
         // Allow derived classes a shot at the sprouts
         if (hd) {
             sproutHook(sproutCollection, d);
         }
+    }
 
-        // Allow derived classes a shot to clean up the parent
-        cleanUpHook(d);
+    /**
+     * Allow derived classes a shot to handle a place exception
+     *
+     * @param hd true if doing heavy-duty processing
+     * @param ex exception thrown by the place
+     *
+     * @return if an error occurred
+     */
+    protected boolean handlePlaceException(boolean hd, Exception ex) {
+        logger.warn("agentProcess {} called from Coordinate problem", (hd ? "HeavyDuty" : "Call"), ex);
+        return true;
+    }
 
-        return sproutCollection;
+    /**
+     * Return true when an error has occurred and the places is configured to not continue
+     *
+     * @param p place that is currently processing the ibdo
+     * @param errorOccurred true if an error occurred
+     *
+     * @return true if processing should not continue
+     */
+    private boolean shouldNotContinueOnError(IServiceProviderPlace p, boolean errorOccurred) {
+        if (allowShouldNotContinueOnError() && errorOccurred) {
+            logger.info("Error terminating coordination step at {}", p);
+            return true;
+        }
+        return false;
+    }
+
+    private void updateTransformHistory(IBaseDataObject d, IServiceProviderPlace p) {
+        if (updateTransformHistory) {
+            DirectoryEntry de = p.getDirectoryEntry();
+            de.setDataType(d.currentForm());
+            // append to the transform history, with flag indicating that the visit was coordinated
+            d.appendTransformHistory(de.getKey(), true);
+        }
+    }
+
+    /**
+     * Process the form
+     *
+     * @param d the ibdo to process
+     * @param errorOccurred true if an error occurred
+     */
+    protected void processForm(IBaseDataObject d, boolean errorOccurred) {
+        if (allowProcessOutputFormOnError() || !errorOccurred) {
+            processOutputForm(d);
+
+            // Clean up my proxies
+            nukeMyProxies(d);
+
+        }
+    }
+
+    /**
+     * If true, process the output form the same for the default and error condition
+     *
+     * @return boolean to allow processing of output for when an error occurs
+     */
+    protected boolean allowProcessOutputFormOnError() {
+        return false;
+    }
+
+    /**
+     * Process the outputs form according to configuration
+     *
+     * @param d the ibdo to process
+     */
+    protected void processOutputForm(IBaseDataObject d) {
```

Comment:
Maybe `applyOutputForm` describes the functionality more specifically than `processOutputForm` does?